File: src/DataStructuresImplementing/list/_LinkedList.java
package DataStructuresImplementing.list;



import DataStructuresImplementing.nodes.JavaNodes.LinkedNode;

public class _LinkedList<T> {
    LinkedNode head;
    public _LinkedList() {
        head= null;
    }
    public void add(Object value){
        LinkedNode newNode=new LinkedNode(value, null,null);
        if(head==null)
            head=newNode;
        else{
            newNode.next=head;
            head.previous=newNode;
            head= newNode;
        }
    }
    public void delete(){
        head=head.next;
        head.previous=null;

    }
    public void display(){
        LinkedNode n=head;
        while(n!=null){
            System.out.println((T)n.value);
            n=n.next;
        }
    }
}

File: src/DataStructuresImplementing/list/_LinkedList.kt
package DataStructuresImplementing.list

import DataStructuresImplementing.nodes.LinkedNode


class LinkedList<T> {
    private var head: LinkedNode<T>? = null

    fun add(value: T) {
        val newNode = LinkedNode(value)
        if (head == null) {
            head = newNode
        } else {
            var current = head
            while (current?.next != null) {
                current = current.next
            }
            current?.next = newNode
        }
    }

    fun printList() {
        var current = head
        while (current != null) {
            print("${current.value} -> ")
            current = current.next
        }
        println("null")
    }
}
